python/pyspark/sql/__init__.py

```diff
@@ -31,9 +31,22 @@
     - L{HiveContext}
       Main entry point for accessing data stored in Apache Hive..
 """
+from __future__ import absolute_import
+
+# fix the module name conflict for Python 3+
```

Comment:
Just to make sure that I understand this conflict, I guess the issue is that Python 3 introduces a `types` module that somehow conflicts with our module?  I think we've discussed this before w.r.t other standard library naming conflicts, but isn't this only a problem if you're trying to import PySpark SQL's `types` module via a relative import rather than absolute one, or if you place the `python/pyspark/sql` folder directly onto your PYTHONPATH?  Is this only a problem during tests?